src/base/listener.spec.ts

```diff
@@ -138,9 +145,11 @@ async function startNode(
     undefined
   )
 
-  await waitUntilListening(listener, new Multiaddr(`/ip4/127.0.0.1/tcp/0/p2p/${peerId.toB58String()}`))
+  process.nextTick(() =>
+    waitUntilListening(listener, new Multiaddr(`/ip4/127.0.0.1/tcp/0/p2p/${peerId!.toB58String()}`))
+  )
```

Comment:
Can you explain what is the technical difference here?

Reply:
Was following the Node.js documentation here ( https://nodejs.org/api/events.html#events_events_on_emitter_eventname_options ) and figured out that the previous version could lead to an unwanted control flow.

The new version (using process.nextTick) attaches the listener (using events.on) and the starts the program flow that generates the events (using process.nextTick) *asynchronously*